roseau/load_flow/models/loads/loads.py

```diff
@@ -221,11 +222,11 @@ def from_dict(cls, data: JsonDict, *, include_results: bool = True) -> "Abstract
             else:
                 fp = None
             self = PowerLoad(id=data["id"], bus=data["bus"], powers=powers, phases=data["phases"], flexible_params=fp)
-        elif (i_list := data.get("currents")) is not None:
-            currents = [complex(i[0], i[1]) for i in i_list]
+        elif load_type == "current":
+            currents = [complex(i[0], i[1]) for i in data["currents"]]
             self = CurrentLoad(id=data["id"], bus=data["bus"], currents=currents, phases=data["phases"])
-        elif (z_list := data.get("impedances")) is not None:
-            impedances = [complex(z[0], z[1]) for z in z_list]
+        elif load_type == "impedance":
+            impedances = [complex(z[0], z[1]) for z in data["impedances"]]
             self = ImpedanceLoad(id=data["id"], bus=data["bus"], impedances=impedances, phases=data["phases"])
         else:
             msg = f"Unknown load type for load {data['id']!r}"
```

Comment:
Since we have the type now
```suggestion
            msg = f"Unknown load type {load_type!r} for load {data['id']!r}"
```